crates/rpcsrv/src/session.rs

```diff
@@ -35,25 +36,49 @@ impl RemoteSession {
         session.get_session_catalog().get_state().as_ref().clone()
     }
 
-    pub async fn execute_serialized_plan(
+    pub async fn create_physical_plan(
         &self,
-        req: ExecuteRequest,
-    ) -> Result<SendableRecordBatchStream> {
-        match req.plan {
-            Some(Plan::LogicalPlan(buf)) => {
-                // TODO: Use a context that actually matters.
-                let fake_ctx = SessionContext::new();
-                let plan = LogicalPlanNode::try_decode(&buf)?
-                    .try_into_logical_plan(&fake_ctx, &GlareDBExtensionCodec)?;
-
-                let mut session = self.session.lock().await;
-                let physical = session.create_physical_plan(plan).await?;
-                let stream = session.execute_physical(physical)?;
-
-                Ok(stream)
-            }
-            Some(Plan::PhysicalPlan(_)) => Err(RpcsrvError::PhysicalPlansNotSupported),
-            None => Err(RpcsrvError::Internal("missing plan on request".to_string())),
-        }
+        logical_plan: impl AsRef<[u8]>,
+    ) -> Result<PhysicalPlanResponse> {
+        let mut session = self.session.lock().await;
+
+        // TODO: Use a context that actually matters.
+        let fake_ctx = SessionContext::new();
+        let plan = LogicalPlanNode::try_decode(logical_plan.as_ref())?
+            .try_into_logical_plan(&fake_ctx, &session.extension_codec())?;
```

Comment:
I would have thought we'd need to change this context with one that had the tables registered. Is handling for the custom table providers done entirely with extension codec?

Reply:
Yeah, we really should change it ASAP. Will create an issue for this.

Handling for all custom providers has to be done through extension codec... I don't see any other way right now 🤔